plugins/ingest/src/main/java/org/elasticsearch/ingest/processor/ConfigurationUtils.java

```diff
@@ -149,4 +149,15 @@ private static String readString(String propertyName, Object value) {
             throw new IllegalArgumentException("property [" + propertyName + "] isn't a map, but of type [" + value.getClass().getName() + "]");
         }
     }
+
+    /**
+     * Returns and removes the specified property as an {@link Object} from the specified configuration map.
+     */
+    public static Object readObject(Map<String, Object> configuration, String propertyName) {
+        if (configuration.containsKey(propertyName)) {
```

Comment:
for the sake of separated error messages, I will treat contains property vs. null property value separately